### PR TITLE
refactor: change underlying clickhouse driver

### PR DESCRIPTION
**Pull request by frank-zsy**
Signed-off-by: frank-zsy <syzhao1988@126.com>

related to #1087 

Upgrade underlying TypeScript/JavaScript client for ClickHouse, update from 20.8 to 22.8.

All notebook has been tested and updated with new client and new server instance, all works fine now.

Upper interface of all metrics does not change at all.

CAUTIONS:

- The result format should be passed into driver as a second parameter rather than simply in the query. Default format is `JSONCompactEachRow` which is the common format used for all metrics. If you want to use the driver directly, you should understand how to set the return format.
- As we may use ClickHouse to store more open source data but not just GHArchive data. The database name changes from `github_log` to `opensource`, and database is set in the client driver on init, so we could only set table name in query SQL. And the tables' names are also changed as below:

| old | new |
|:--:|:--:|
| events | gh_events |
| export_repo | gh_export_repo |
| export_user | gh_export_user |
| repo_openrank | gh_repo_openrank |
| user_openrank | gh_user_openrank |

We may add more data about open source supply chain and artifact data to ClickHouse server later.

### CODE REVIEW

**Comment by frank-zsy** [comment]
This PR is only used to upgrade the kernel scripts part, sample data upgrade will be submitted later.

File: src/db/clickhouse.ts [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 
-export async function queryStream<T = any>(q: string, onRow: (row: T) => void): Promise<void> {
+export async function queryStream<T = any>(q: string, onRow: (row: T) => void, options: any = {}): Promise<void> {
   return new Promise(async resolve => {
-    const stream = (await getClient()).query(q);
-    stream.on('data', (row: T) => onRow(row));
-    stream.on('end', () => {
-      resolve();
-    });
+    const resultSet = await (await getClient()).query({ query: q, format: 'JSONCompactEachRow', ...options });
```
Comment (xgdyp) [inline src/db/clickhouse.ts:19]: So in python kernel, what we need to change is table name ,db engine and `FORMAT JSONCompact` like this format, is this right?

Comment (frank-zsy) [reply src/db/clickhouse.ts:19]: Nope, this is JavaScript kernel with JavaScript client, for Python kernel, you guys need to test it, if current Python kernel client is compatible with the new ClickHouse server version, you don't need to change anything. You only need to change Python code if the client is not compatible with new server.

Keep in mind that: JavaScript/TypeScript kernel code and Python kernel code are totally independent.

Comment (frank-zsy) [reply src/db/clickhouse.ts:19]: But for sure, you need to change the database and table name for queries.

Comment (xgdyp) [reply src/db/clickhouse.ts:19]: OK

**Review by xgdyp — APPROVED** [review]

**Comment by frank-zsy** [comment]
@xgdyp You can approve this PR if you think it is fine. There is no need to get all reviewers to approve, OpenDigger only requires 1 reviewer indeed.

**Pull request merged**